File: app/src/main/java/org/neidhardt/dynamicsoundboard/dialog/soundsettings/SoundSettingsBaseDialog.java
package org.neidhardt.dynamicsoundboard.dialog.soundsettings;

import android.app.Dialog;
import android.os.Bundle;
import org.neidhardt.dynamicsoundboard.dialog.BaseDialog;
import org.neidhardt.dynamicsoundboard.mediaplayer.EnhancedMediaPlayer;

/**
 * Created by eric.neidhardt on 12.04.2015.
 */
public abstract class SoundSettingsBaseDialog extends BaseDialog
{
	private static final String KEY_PLAYER_ID = "org.neidhardt.dynamicsoundboard.dialog.soundsettings.SoundSettingsBaseDialog.playerId";
	private static final String KEY_FRAGMENT_TAG = "org.neidhardt.dynamicsoundboard.dialog.soundsettings.SoundSettingsBaseDialog.fragmentTag";

	protected String fragmentTag;
	protected EnhancedMediaPlayer player;

	protected static void addArguments(BaseDialog dialog, String playerId, String fragmentTag)
	{
		Bundle args = new Bundle();
		args.putString(KEY_PLAYER_ID, playerId);
		args.putString(KEY_FRAGMENT_TAG, fragmentTag);

		dialog.setArguments(args);
	}

	@Override
	public void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		Bundle args = this.getArguments();
		if (args != null)
		{
			String playerId = args.getString(KEY_PLAYER_ID);
			this.fragmentTag = args.getString(KEY_FRAGMENT_TAG);
			this.player = this.getServiceManagerFragment().getSoundService().searchForId(this.fragmentTag, playerId);
		}
	}

	@Override
	public abstract Dialog onCreateDialog(Bundle savedInstanceState);
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundmanagement/dialog/SoundSettingsBaseDialog.kt
package org.neidhardt.dynamicsoundboard.soundmanagement.dialog

import android.app.Dialog
import android.os.Bundle
import org.neidhardt.dynamicsoundboard.SoundboardApplication
import org.neidhardt.dynamicsoundboard.mediaplayer.MediaPlayerController
import org.neidhardt.dynamicsoundboard.views.BaseDialog

/**
 * File created by eric.neidhardt on 12.04.2015.
 */
abstract class SoundSettingsBaseDialog : BaseDialog()
{
	private val soundsDataAccess = SoundboardApplication.soundsDataAccess

	protected abstract var fragmentTag: String
	protected abstract var player: MediaPlayerController

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val args = this.arguments
		if (args != null) {
			val playerId = args.getString(KEY_PLAYER_ID)
			this.fragmentTag = args.getString(KEY_FRAGMENT_TAG)
			this.player = this.soundsDataAccess.getSoundById(this.fragmentTag, playerId) as MediaPlayerController
		}
	}

	abstract override fun onCreateDialog(savedInstanceState: Bundle?): Dialog

	companion object {
		private val KEY_PLAYER_ID = "org.neidhardt.dynamicsoundboard.soundmanagement.dialog.SoundSettingsBaseDialog.playerId"
		private val KEY_FRAGMENT_TAG = "org.neidhardt.dynamicsoundboard.soundmanagement.dialog.SoundSettingsBaseDialog.fragmentTag"

		fun addArguments(dialog: SoundSettingsBaseDialog, playerId: String, fragmentTag: String) {
			val args = Bundle()
			args.putString(KEY_PLAYER_ID, playerId)
			args.putString(KEY_FRAGMENT_TAG, fragmentTag)

			dialog.arguments = args
		}
	}
}
